### PR TITLE
api implemented

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**
Hello, i have Added GET ALL POSTS, GET POST ACC TO USER, SEE USERS PROFILE, SEE ANOTHER USER PROFILE, SINGLE POST  functionality, in social media app. kindly review it and let me know what do think about it !
files to check:
src/app/slice/postSlice.js
src/app/slice/userSlice.js
src/pages/FeedDetailsPage/FeedDetailsPage.js
src/pages/Shared/Feed/Feed.js
src/pages/UserProfile/ProfileDetails.js
src/pages/UserProfile/UserProfile.js
PREVIEW: https://blithe-in.netlify.app

### CODE REVIEW

File: src/Helpers/Services/actions.jsx [hunk 1]
```diff
@@ -25,3 +25,59 @@ export const actionSignup = async (data) => {
...
+    return response;
+    }else{
+      console.log("response.data.errors:", response.data.errors);
+      // showErrorToast(response.data.errors[0])
+  }
+};
+
+//action to get All users
+export const getUserById = async (data) => {
+  const response = await mainApiService("getUserById", data);
```
Comment (mehulsatardekar) [inline src/Helpers/Services/actions.jsx:43]: use try-catch block to handle api response and errors

File: src/pages/MainPage/MainPageRoute.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ import RouteData from './RouteData'
 const MainPageRoute = () =>{
 
     return(
-        // <Suspense fallback={<Loader/>}>
-        <Suspense fallback={<div>Loading...</div>}>
+        <Suspense fallback={<Loader/>}>
+         {/* <Suspense fallback={<div>Loading...</div>}> */}
             <RouteData/>
```
Comment (mehulsatardekar) [inline src/pages/MainPage/MainPageRoute.js:11]: here you have wrapped RouteData Component inside suspense but you haven't imported RouteData as React lazy.

i mean this should be const RouteData = React.lazy(()=> import ('./PrivateRoute'));  

you can refer to this article
  --- https://17.reactjs.org/docs/concurrent-mode-suspense.html

File: src/pages/MainPage/RouteData.js [hunk 3]
```diff
@@ -20,15 +23,15 @@ function NoMatch() {
 const RouteData = () =>{
     return(
         <Routes>  
-           {/* <Route exact path='/user' element={<PrivateRoute/>}>                
-              <Route exact path="/user/liked-videos"  element={<LikedVideosPage/>}/> 
-              <Route exact path='/user/playlists' element={<PlayList/>}/>
-              <Route exact path='/user/watchlater' element={<WatchLater/>}/>
-              <Route exact path='/user/history' element={<History/>}/>
+           <Route exact path='/user' element={<PrivateRoute/>}>                
+              {/* <Route exact path="/user/profile/:id"  element={<UserProfile/>}/>  */}
```
Comment (mehulsatardekar) [inline src/pages/MainPage/RouteData.js:27]: here you are doing code splitting on route base level. but you haven't wrapped component under suspense
eg:

`**<suspense fallback={//fallback component goes here}>//component goes here</suspense>**`

File: src/pages/Shared/Loader/Loader.css [hunk 4]
```diff
@@ -1,3 +1,24 @@
+.parentDisable{
+  position: fixed;
+  top: 0;
+  left: 0;
+  background: #666;
+  opacity: 0.8;
+  z-index: 9999999999;
```
Comment (mehulsatardekar) [inline src/pages/Shared/Loader/Loader.css]: you can lower down the z index values

File: src/pages/UserProfile/ProfileDetails.js [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+                  <Stepper orientation="vertical">
+                    {userDetailsData?.workExperience.map((experience, index) => (
+                      <Step key={index} active={false} expanded={true}>
+                        <StepLabel
+                          icon={<CircleIcon />}
+                          className="step-label"
+                          id="step-label"
+                          // optional={
+                          //     index === 2 ? (
+                          //     <Typography variant="caption">Last step</Typography>
```
Comment (mehulsatardekar) [inline src/pages/UserProfile/ProfileDetails.js]: remove commented code

File: src/pages/UserProfile/ProfileDetails.js [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+            </Typography>
+            <Box className="experience-details-box">
+              {/* <Box className="experience-item"> */}
+                {/* <PeopleAltIcon /> */}
+                {/* <Box>
+                  <Typography variant="body1">
+                    Software Developer(react js)
+                  </Typography>
+                  <Typography variant="body2">
+                    LMS Solutions (India) Pvt. Ltd. · Full-time
```
Comment (mehulsatardekar) [inline src/pages/UserProfile/ProfileDetails.js]: remove commented code

**Review by mehulsatardekar — APPROVED** [review]

File: src/app/slice/postSlice.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getAllPosts, getAllUsers, getPostById, getPostByUsername, getUserById } from "../../Helpers/Services/actions"; 
+// import { updateUser } from "../Auth/authSlice";
+
+const initialState = {
```
Comment (mikesch803) [inline src/app/slice/postSlice.js:5]: you havent taken initial state ie post []

File: src/app/slice/postSlice.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    }
+  );
+
+  const postSlice = createSlice({
+    name: "post",
+    initialState,
+    reducers: {},
+    extraReducers: {
+        [getPosts.pending]: (state, action) => {
+        state.posts = {
```
Comment (mikesch803) [inline src/app/slice/postSlice.js:54]: this state.posts is not in initial state

**Review by mikesch803 — APPROVED** [review]
i commented at some lines where i m not getting, apart from it ui is nice, mobile responsive, mentioned apis also working, overall god job!

**Review by MayankKumar10 — APPROVED** [review]
Remove unnecessary comments everything else looks good Nice UI you can merge it.

File: src/pages/MainPage/MainPageRoute.js (hunk 2 above)
Comment (kgajriaAgarwal) [reply src/pages/MainPage/MainPageRoute.js:11]: image
Done lazy routing in this manner of routes

File: src/pages/MainPage/RouteData.js (hunk 3 above)
Comment (kgajriaAgarwal) [reply src/pages/MainPage/RouteData.js:27]: will follow the above mentioned structure 
image

File: src/pages/Shared/Loader/Loader.css (hunk 4 above)
Comment (kgajriaAgarwal) [reply src/pages/Shared/Loader/Loader.css]: done!

File: src/pages/UserProfile/ProfileDetails.js (hunk 5 above)
Comment (kgajriaAgarwal) [reply src/pages/UserProfile/ProfileDetails.js]: removed!

File: src/pages/UserProfile/ProfileDetails.js (hunk 6 above)
Comment (kgajriaAgarwal) [reply src/pages/UserProfile/ProfileDetails.js]: removed!

File: src/app/slice/postSlice.js (hunk 7 above)
Comment (kgajriaAgarwal) [reply src/app/slice/postSlice.js:5]: managed using state.posts

File: src/app/slice/postSlice.js (hunk 8 above)
Comment (kgajriaAgarwal) [reply src/app/slice/postSlice.js:54]: ok, managed using state.posts

**Pull request merged**